### PR TITLE
Various changes

### PR DESCRIPTION
**Pull request by Jukspa**
hf

### CODE REVIEW

File: spt/OrangeBox/modules/vguimatsurfaceDLL.cpp [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+#include "..\scripts\srctas_reader.hpp"
+#include "..\..\vgui\vgui_utils.hpp"
+#include "vgui_controls\controls.h"
+#include "..\patterns.hpp"
+#include "..\overlay\portal_camera.hpp"
+#include "..\..\utils\ent_utils.hpp"
+#include "..\..\utils\property_getter.hpp"
+#include "..\module_hooks.hpp"
+
+#ifndef OE
```
Comment (YaLTeR) [inline spt/OrangeBox/modules/vguimatsurfaceDLL.cpp]: Either leave this and get rid of the ones in the function bodies or the other way around.

File: spt/OrangeBox/modules/vguimatsurfaceDLL.cpp [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+		swprintf_s(buffer, BUFFER_SIZE, L"ag sg: %s", result.c_str());
+		DRAW();
+	}
+#endif
+
+	if (y_spt_hud_oob.GetBool())
+	{
+		Vector v = clientDLL.GetPlayerEyePos();
+		trace_t tr;
+		Strafe::Trace(tr, v, v + Vector(1,1,1), Strafe::HullType::POINT);
```
Comment (YaLTeR) [inline spt/OrangeBox/modules/vguimatsurfaceDLL.cpp]: What if you trace from v to v without adding arbitrary 1,1,1?

File: spt/OrangeBox/modules/vguimatsurfaceDLL.cpp [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+
+		bool oob = engineDLL.ORIG_CEngineTrace__PointOutsideWorld(nullptr, 0, v) && !tr.startsolid;
```
Comment (YaLTeR) [inline spt/OrangeBox/modules/vguimatsurfaceDLL.cpp]: This sounds like it'll report inbounds even when part of the player is sticking out into oob.

File: spt/OrangeBox/spt-serverplugin.cpp [hunk 4]
```diff
@@ -568,9 +610,48 @@ CON_COMMAND(y_spt_canjb, "Tests if player can jumpbug on a given height, with th
...
+
+	if (can.canJB)
+		Msg("Yes, projected landing height %.6f in %d ticks\n", can.landingHeight - height, can.ticks);
+	else
+		Msg("No, missed by %.6f in %d ticks.\n", can.landingHeight - height, can.ticks);
+}
+
+#ifndef OE
+
+CON_COMMAND(y_spt_print_ents, "Prints all client entity indexes and their corresponding classes.")
```
Comment (YaLTeR) [inline spt/OrangeBox/spt-serverplugin.cpp]: indices

File: spt/utils/patterncontainer.hpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <map>
+#include <string>
+#include <vector>
+#include "SPTLib\memutils.hpp"
+
+#define PVOID void*
```
Comment (YaLTeR) [inline spt/utils/patterncontainer.hpp]:
```suggestion
using PVOID = void*;
```

File: spt/utils/string_parsing.cpp [hunk 6]
```diff
@@ -1,5 +1,24 @@
 #include "stdafx.h"
+#include <Windows.h>
 #include "string_parsing.hpp"
+#include <algorithm>
+
+std::wstring s2ws(const std::string& s)
```
Comment (YaLTeR) [inline spt/utils/string_parsing.cpp]: SPTLib has this, it's called `Convert()`

File: Tests/README.txt [hunk 7]
```diff
@@ -1,6 +1,6 @@
 How to run tests:
 
-Install python(any python 3 will probably work) and put it in your PATH env variable.
-Set up your config.ini to include all the build directories you have(see format in config_example.ini).
-Run the tests with the command: python run_tests.py > output.txt. The games will be automatically started.
+Install Python(any Python 3 will probably work) and put it in your PATH env variable.
```
Comment (YaLTeR) [inline Tests/README.txt]:
```suggestion
Install Python (any Python 3 will probably work) and put it in your PATH env variable.
```

File: spt/OrangeBox/modules.hpp [hunk 8]
```diff
@@ -16,4 +16,7 @@ extern VGui_MatSurfaceDLL vgui_matsurfaceDLL;
 #define TRACE_MSG(X) DevMsg(X)
 #else
 #define TRACE_MSG(X)
- #endif
\ No newline at end of file
+#endif
+
+#define TRACE_ENTER() TRACE_MSG("ENTER:" __func__ "\n")
+#define TRACE_EXIT() TRACE_MSG("EXIT:" __func__ "\n")
```
Comment (YaLTeR) [inline spt/OrangeBox/modules.hpp]:
```suggestion
#define TRACE_EXIT() TRACE_MSG("EXIT: " __func__ "\n")
```

File: spt/OrangeBox/modules.hpp [hunk 9]
```diff
@@ -3,7 +3,20 @@
...
+extern VGui_MatSurfaceDLL vgui_matsurfaceDLL;
+#endif
+
+#ifdef TRACE
+#define TRACE_MSG(X) DevMsg(X)
+#else
+#define TRACE_MSG(X)
+#endif
+
+#define TRACE_ENTER() TRACE_MSG("ENTER:" __func__ "\n")
```
Comment (YaLTeR) [inline spt/OrangeBox/modules.hpp]:
```suggestion
#define TRACE_ENTER() TRACE_MSG("ENTER: " __func__ "\n")
```

File: spt/utils/patterncontainer.hpp (hunk 5 above)
Comment (mikesmiffy128) [reply spt/utils/patterncontainer.hpp]: I'm too late xd but in general is it really better to typedef this vs just using `void *`?

Comment (YaLTeR) [reply spt/utils/patterncontainer.hpp]: I thought about commenting about this but then I was like eh

**Pull request merged**